### DISCUSSION TITLE
Create wt -info-vt (or similar) to print list of supported VT sequences

### DISCUSSION THREAD

**Discussion by WSLUser** [microsoft/terminal#13719] [Ideas]
# Description of the new feature/enhancement
Just like the terminfo file can be auto-generated, or the VT parser FSM, it is also possible to auto-generate the list of supported VT sequences along with some additional info. Expose the list of supported VT sequences via CLI.

Allow users and terminal applications to easily determine WT VT features without need of a terminfo.
# Proposed technical implementation details (optional)
Contour has done this and has a pretty nice result. However Contour relies on using Crispy, a dependency it has. See https://github.com/contour-terminal/contour/pull/731

**Comment by zadjii-msft** [comment]
This just sounds like creating _another standard_. What value does this provide over just a (well maintained) `terminfo` entry/? 

Windows Terminal's got such a complicated VT support story (split across conhost/conpty/WT itself) that I just feel like this provides a minimal amount of value here for the complexity we'd have to deal with. We don't even the bandwidth to keep a `terminfo` entry maintained. Heck, even our own VT docs are pretty poorly maintained, despite us having a script that (at one point) would auto generate a table for us. 

Not to mention, `wt.exe` is a WINDOWS process that wouldn't be able to print to the console it's called from. So the ergonomics here of printing that text to a win32 `MessageBox` (or similar) seems... bad. 


I just don't see what extra value this gets us. Happy to be proven wrong though.

**Comment by WSLUser** [comment]
There are plenty of terminal applications that are using other heuristics other than terminfo to determine terminal features. You don't even have to look any further than tmux for proof. This isn't about another standard, this is just helping applications have an easier time knowing what they can support on a given terminal without doing some of the many hacky things they already are to support users. Also the idea here is to make it so we can keep that documentation updated as we do keep getting out of date in reporting available features. A big consensus I'm seeing among certain popular terminal applications is to not rely on terminfo at all and simply acquire the information on sequence support through other means. This would be one of those means. Notcurses for example does a ton of trickery to determine things for every terminal it knows about because terminfo isn't reliable enough to ensure features are reported. Maybe it should be but it isn't and many features just aren't getting reported there. So even if this does become a standard, it's a far better one than terminfo at this point in the game.

**Comment by WSLUser** [comment]
The challenges regarding wt.exe are concerning but I was hopeful we could re-use our existing arg parser as it exists today without needing to introduce another one that sounds like it would be supporting Linux mechanisms through WSL or a VM or something, which I don't think would be what we want here. It should be native to Windows itself. Contour provides an excellent example but Contour is also cross-platform. WT is not so we need to determine the best way to use what Windows has available (or make it available in conhost) to report this information in the terminal window.

**Comment by zadjii-msft** [comment]
Frankly, I'd rather do something more standardized across multiple emulators, than just adding another ad-hoc solution to further muddle the problem space. I'm gonna convert this thread to a discussion, and we can continue planning there until a better consensus is achieved.
